src/components/ParticipantTracks/ParticipantTracks.tsx

```diff
@@ -8,7 +8,7 @@ interface ParticipantTracksProps {
   participant: LocalParticipant | RemoteParticipant;
   disableAudio?: boolean;
   enableScreenShare?: boolean;
-  videoPriority?: Track.Priority;
+  videoPriority?: Track.Priority | null;
```

Comment:
Can you also explain the rationale behind this? Does this mean that when a video track is no longer at the focus it gets assigned 'null' as its subscribe priority?

Reply:
When `videoPriority` is `null`, `track.setPriority` never gets called. 

I only added this change because it seems better to set `videoPriority` to `null` instead of `undefined`.

Reply:
@timmydoza We should find a way to differentiate between a VideoTrack that was added to the main window due to a dominant speaker change event vs the VideoTrack that was added to the main window due to the user pinning a Participant. In the former case, we should not be calling `setPriority()` at all. In the latter case, we should call `setPriority('high')` when pinned and `setPriority(null)` when unpinned.

Reply:
@Sounds good. I've updated the logic so that the priority is set to high only when the main participant is not the dominant speaker or sharing their screen (because screen share tracks are already published with high priority).  `setPriority(null)` will be only be called when pinned participants are unpinned.

Reply:
@timmydoza one question that comes to may mind is what happens with the screenshare when the end-user pins another video. If I'm not wrong, the screenshare is published with priority 'high', which is OK because buy default it goes to the focus (i.e. the main area of the screen). However, if an end-user pins a different video, the screenshare should be downgraded to 'low' given that it's not being rendered. If we keep it as 'high', then it will steal relevant BW (and quality) to the pinned video while it's pinned. Is this how things work in this patch?

Reply:
Thats correct. We used to have screen share tracks published with low priority, and then set their priority to high while they are displayed in the main video area. 

We then set the screen share tracks to always have high priority to address an issue found in this ticket: https://issues.corp.twilio.com/browse/VMS-2462. Users were reporting that screenshare tracks were getting switched off. 

It seems that the reported issue was due to a `setPriority` bug in the JS SDK which was fixed in version 2.3.0.  

I'm thinking we go back to how we used to handle screen share priority.  We publish screen share tracks with low priority, and then set the priority to high when its displayed in the main video area. What do you think @lwilio?

Reply:
@timmydoza If the bug is fixed IMO we should do it the right way and set the screenshare priority to low when it's not visible. Otherwise we'll be spoiling valuable BW for nothing. Of course, please double check first that the bug is really fixed just in case.

Reply:
@lwilio It appears to be fixed: https://github.com/twilio/twilio-video.js/blob/master/CHANGELOG.md#bug-fixes

Should I 
1) publish screen share tracks with 'low' priority and set them to 'high' when they are rendered in the main video area?
or
2) publish screen share tracks with 'high' priority, and set them to 'low' when they are not rendered in the main video area?

I'd very much prefer to go with #1 as it is much simpler to code, but I'm curious about what you think.

Reply:
@timmydoza IMO there will be currently no difference between 1 and 2 from the QoE perspective. However, I'd prefer 2 because in the future track priorities would also be considered at the publisher side (currently are only considered at the subscriber side). Also, the default behavior is to show the screenshare in the focus, hence, IMO it's clear if it's set to 'high' "by default" and we only set it to low when something "special" happens (i.e. the end-user pins another video). However, again, up to you to choose.

Reply:
Thanks for your input @lwilio. I've thought about this one a lot, and for now I'd like to go with option #1. It's easier and faster to implement. I'm in favor of trying option 2 in the near future.